test/functionalTest/cases/3162_context_providers_legacy_non_primitive/query_for_arrays.test

```diff
@@ -23,15 +23,13 @@
 #
```

Comment:
In fact... appart from "format changes" (e.g. echo, content-length adjust, JSON pretty print, etc.) is there any real change in the PR?

Reply:
The real change is that instead of just copying the compound pointer in `ContextAttribute::ContextAttribute(ContextAttribute* caP, bool useDefaultType)`,
the compound is cloned.

Reply:
This change gave leaks that were ... really hard to understand.
My guess: timing.
To understand the leaks I saved a pointer of all compounds cloned right there in "the real change", and on free I remove those pointers, to finally free those that were still in the vector.

[ I just remembered this ] Imagine my surprise when finding that there were no compounds left to free.

However, if I remove the saving-compound-pointers-to-vector, then I get memory leaks.
So, the vector is "unneeded code" that is essential due to the timing it changes. Without it, we get memory leaks.

And now you know why I don't like this fix ...

It works, but more time for investigation is needed. Doesn't have to be right now. Especially not as this fix is needed.

Reply:
Ups... I didn't make myself clear enough. I mean, not the changes in the PR in general, but the changes of the .test in the PR, i.e.:

> In fact... appart from "format changes" (e.g. echo, content-length adjust, JSON pretty print, etc.) is there any real change in **this .test**?

Reply:
(The explanation about the PR itself is good... but I'll move to the general comment in the PR to continue discussion)